sdk/textanalytics/Azure.AI.TextAnalytics/src/TextAnalyticsClient.cs

```diff
@@ -2072,6 +2072,332 @@ private Response<RecognizeLinkedEntitiesResultCollection> RecognizeLinkedEntitie
+        // Look for other alternatives of operation cancellation
+        #pragma warning disable AZC0003 // DO make service methods virtual.
+        #pragma warning disable AZC0015 // Unexpected client method return type.
+        public virtual async Task<string> StartCancelHealthJobAsync(HealthcareOperation operation, CancellationToken cancellationToken = default)
+        #pragma warning restore AZC0015 // Unexpected client method return type.
+        #pragma warning restore AZC0003 // DO make service methods virtual.
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TextAnalyticsClient)}.{nameof(StartCancelHealthJobAsync)}");
+            scope.Start();
+
+            try
+            {
+                ResponseWithHeaders<TextAnalyticsCancelHealthJobHeaders> response = await _serviceRestClient.CancelHealthJobAsync(new Guid(operation.Id), cancellationToken).ConfigureAwait(false);
+
+                return response.Headers.OperationLocation;
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+
+        }
+
+        /// <summary> Cancel healthcare prediction job. </summary>
+        /// <param name="operation"> Job ID. </param>
+        /// <param name="cancellationToken"> The cancellation token to use. </param>
+        #pragma warning disable AZC0003 // DO make service methods virtual.
+        #pragma warning disable AZC0015 // Unexpected client method return type.
+        public virtual string StartCancelHealthJob(HealthcareOperation operation, CancellationToken cancellationToken = default)
+        #pragma warning restore AZC0015 // Unexpected client method return type.
+        #pragma warning restore AZC0003 // DO make service methods virtual.
+        {
+            using DiagnosticScope scope = _clientDiagnostics.CreateScope($"{nameof(TextAnalyticsClient)}.{nameof(StartCancelHealthJobAsync)}");
+            scope.Start();
+
+            try
+            {
+                ResponseWithHeaders<TextAnalyticsCancelHealthJobHeaders> response = _serviceRestClient.CancelHealthJob(new Guid(operation.Id), cancellationToken);
+
+                return response.Headers.OperationLocation;
+            }
+            catch (Exception e)
+            {
+                scope.Failed(e);
+                throw;
+            }
+        }
+
+        /// <summary>
+        /// Gets collection of healthcare entities from the HealthOperation using async pageable.
+        /// </summary>
+        /// <param name="operation"></param>
+        /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+        /// <returns>A collection of <see cref="RecognizeHealthcareEntitiesResultCollection"/> items.</returns>
+        public virtual AsyncPageable<DocumentHealthcareResult> GetHealthcareEntities(HealthcareOperation operation, CancellationToken cancellationToken = default)
```

Comment:
I can't find where this method is used.
what is the purpose of it?

Reply:
This returns all the entities by scrolling over all the nextLinks. Updated sample and test.

Reply:
ahh I see. so the pagination is done here.... It will be better to have the pagination in the `HealthCareOperation.Update` method so that it is transparent for the user that they only need to do `client.StartHelathcareEntities(...)` instead of them knowing that if they send more than 20 documents they need to call this method too.